contracts/integrations/test/internal-integration-tests/exchange_wrapper_test.ts

```diff
@@ -0,0 +1,1058 @@
+    IExchangeFillEventArgs,
+    LocalBalanceStore,
+} from '@0x/contracts-exchange';
+import { ReferenceFunctions } from '@0x/contracts-exchange-libs';
+import { artifacts as stakingArtifacts } from '@0x/contracts-staking';
+import {
+    blockchainTests,
+    constants,
+    describe,
+    expect,
+    getLatestBlockTimestampAsync,
+    Numberish,
+    TransactionHelper,
+    verifyEvents,
+} from '@0x/contracts-test-utils';
+import { assetDataUtils, ExchangeRevertErrors, orderHashUtils } from '@0x/order-utils';
+import { FillResults, OrderStatus, SignedOrder } from '@0x/types';
+import { BigNumber } from '@0x/utils';
+import { Web3Wrapper } from '@0x/web3-wrapper';
+import { TransactionReceiptWithDecodedLogs } from 'ethereum-types';
+import * as _ from 'lodash';
+
+import { Actor } from '../actors/base';
+import { Maker } from '../actors/maker';
+import { DeploymentManager } from '../utils/deployment_manager';
+
+const { addFillResults, safeGetPartialAmountFloor } = ReferenceFunctions;
+
+// tslint:disable:no-unnecessary-type-assertion
+blockchainTests.resets('Exchange wrappers', env => {
+    let chainId: number;
+    let maker: string;
+    let taker: string;
+    let feeRecipient: string;
+
+    let makerActor: Maker;
+    let takerActor: Actor;
+
+    const nullFillResults: FillResults = {
+        makerAssetFilledAmount: constants.ZERO_AMOUNT,
+        takerAssetFilledAmount: constants.ZERO_AMOUNT,
+        makerFeePaid: constants.ZERO_AMOUNT,
+        takerFeePaid: constants.ZERO_AMOUNT,
+        protocolFeePaid: constants.ZERO_AMOUNT,
+    };
+
+    let deployment: DeploymentManager;
+    let blockchainBalances: BlockchainBalanceStore;
+    let initialLocalBalances: LocalBalanceStore;
+    let localBalances: LocalBalanceStore;
+
+    let wethAssetData: string;
+    let txHelper: TransactionHelper;
+
+    before(async () => {
+        chainId = await env.getChainIdAsync();
+        [feeRecipient] = await env.getAccountAddressesAsync();
+
+        deployment = await DeploymentManager.deployAsync(env, {
+            numErc20TokensToDeploy: 3,
+            numErc721TokensToDeploy: 0,
+            numErc1155TokensToDeploy: 0,
+        });
+
+        makerActor = new Maker({
+            name: 'market maker',
+            deployment,
+            orderConfig: {
+                ...constants.STATIC_ORDER_PARAMS,
+                makerAssetData: assetDataUtils.encodeERC20AssetData(deployment.tokens.erc20[0].address),
+                takerAssetData: assetDataUtils.encodeERC20AssetData(deployment.tokens.erc20[1].address),
+                makerFeeAssetData: assetDataUtils.encodeERC20AssetData(deployment.tokens.erc20[2].address),
+                takerFeeAssetData: assetDataUtils.encodeERC20AssetData(deployment.tokens.erc20[2].address),
+                feeRecipientAddress: feeRecipient,
+                exchangeAddress: deployment.exchange.address,
+                chainId,
+            },
+        });
+        maker = makerActor.address;
```

Comment:
optional nit: I've been naming the `Actor`s "maker" and just using `maker.address` in tests